lib/elixir/lib/file/stream.ex

```diff
@@ -36,7 +36,7 @@ defmodule File.Stream do
 
   defimpl Collectable do
     def into(%{path: path, modes: modes, raw: raw} = stream) do
-      modes = for mode <- modes, not mode in [:read], do: mode
+      modes = modes -- [:read]
```

Comment:
That's an unrelated change, but I think it simplifies the code and is small enough that it's ok to merge them together.